ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/RLPInput.java

```diff
@@ -350,7 +350,7 @@ default long readUnsignedInt() {
    */
   default <T> List<T> readList(final Function<RLPInput, T> valueReader) {
     final int size = enterList();
-    final List<T> res = new ArrayList<>(size);
+    final List<T> res = size == 0 ? List.of() : new ArrayList<>(size);
```

Comment:
This is a way to reduce the memory usage, and it is related to the txpool in the sense that less memory a tx uses the more tx we can fit in the same space, there are also other optimization like this one and more potential todo, and I am now thinking that it could make sense to move them in a separate PR since this is already big enough